.github/workflows/release.yml

```
@@ -0,0 +1,76 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - v[0-9]+.*
+  workflow_dispatch:
+
+env:
+  RUST_BACKTRACE: 1
+  CARGO_TERM_COLOR: always
+
+jobs:
+  create-release:
+    name: Create release
+    runs-on: ubuntu-latest
```

I'd add another safety check here to make sure this is all running on the `main` branch and wasn't accidentally called from somewhere else:
```suggestion
    if: github.ref == 'refs/heads/main'
    runs-on: ubuntu-latest
```

---

Well, the ref here is not main, so that's a bit problematic

---

If using a workflow-to-workflow call as I suggested, it would be main since the workflow wouldn't be triggered on a tag push.